Assets/Scripts/Models/WorldGenerator.cs

```diff
@@ -102,6 +102,16 @@ public static void Generate(World world, int seed)
 
                                 if (randomweight <= currentweight)
                                 {
+                                    
+                                    if (inv.objectType == "Raw Iron" || inv.objectType == "Uranium")
+                                    {
+                                        Furniture mine = PrototypeManager.Furniture.GetPrototype("mine");
```

Comment:
add `.Clone();` to the end and this will not be prototype anymore.
And create overload function in World

```
public Furniture PlaceFurniture(string objectType, Tile t, bool doRoomFloodFill = true)
    {
        // TODO: This function assumes 1x1 tiles -- change this later!
        if (PrototypeManager.Furniture.HasPrototype(objectType) == false)
        {
            Debug.ULogErrorChannel("World", "furniturePrototypes doesn't contain a proto for key: " + objectType);
            return null;
        }

        Furniture furn = Furniture.PlaceInstance(PrototypeManager.Furniture.GetPrototype(objectType), t);

        return PlaceFurniture(furn, t, doRoomFloodFill );
    }

public Furniture PlaceFurniture( Furniture furniture, Tile t, bool doRoomFloodFill = true)
    {
        if (furn == null)
        {
            // Failed to place object -- most likely there was already something there.
            return null;
        }

        furn.Removed += OnFurnitureRemoved;
        furnitures.Add(furn);

        // Do we need to recalculate our rooms?
        if (doRoomFloodFill && furn.RoomEnclosure)
        {
            Room.DoRoomFloodFill(furn.Tile);
        }

        if (OnFurnitureCreated != null)
        {
            OnFurnitureCreated(furn);

            if (furn.MovementCost != 1)
            {
                // Since tiles return movement cost as their base cost multiplied
                // buy the furniture's movement cost, a furniture movement cost
                // of exactly 1 doesn't impact our pathfinding system, so we can
                // occasionally avoid invalidating pathfinding graphs.
                // InvalidateTileGraph();    // Reset the pathfinding system
                if (tileGraph != null)
                {
                    tileGraph.RegenerateGraphAtTile(t);
                }
            }
        }

        return furn;
    }
```